### PR TITLE
Fix `PostgreSQL` grants

### LINKED ISSUE #5129: check in MigrationManager.php not compatible with postgresql
## Describe the bug
In the web interface I get an error
SQLSTATE[22023]: Invalid parameter value: 7 ERROR: unrecognized privilege type: "CREATE", query was: SELECT has_table_privilege(:user, :table, :privilege) AS table_privilege_granted
PostgreSQL doesn't have the CREATE privilege.

In the postgres log I see: 
`2023-09-27 09:09:54.232 UTC [4154218] icinga2@icinga2 ERROR:  unrecognized privilege type: "CREATE"
2023-09-27 09:09:54.232 UTC [4154218] icinga2@icinga2 STATEMENT:  SELECT has_table_privilege($1, $2, $3) AS table_privilege_granted
`

I removed 'CREATE' in the  public function getRequiredDatabasePrivileges() in /usr/share/php/Icinga/Application/MigrationManager.php
return ['CREATE','SELECT','INSERT','UPDATE','DELETE','DROP','ALTER','CREATE VIEW','INDEX','EXECUTE'];
-> return ['SELECT','INSERT','UPDATE','DELETE','DROP','ALTER','CREATE VIEW','INDEX','EXECUTE'];
which fixed the problem for me. I leave it up to someone else to find a more generic solution.

## To Reproduce

1.In the web interface an error occurs when clicking on the settings icon -> system -> migrate

## Expected behavior
Status screen

## Screenshots
image

## Your Environment
Include as many relevant details about the environment you experienced the problem in

* Icinga Web 2 version and modules (System - About): 2.12.0
* Web browser used: Firefox
* Icinga 2 version used (`icinga2 --version`): r2.14.0-1
* PHP version used (`php --version`): 8.1.2-1ubuntu2.14
* Server operating system and version: Ubuntu 22.04.3 LTS
*  Database: Postgres 14

## Additional context
--

### PR DESCRIPTION
**Pull request by yhabteab**
fixes #5129

### CODE REVIEW

**Review by nilmerg — CHANGES_REQUESTED** [review]
Still need to test this

File: library/Icinga/Application/MigrationManager.php [hunk 1]
```diff
@@ -334,13 +336,20 @@ protected function checkRequiredPrivileges(
...
+
+        $isPgsql = $conn->getAdapter() instanceof Sql\Adapter\Pgsql;
+        $privileges = $this->getRequiredDatabasePrivileges();
+        $dbPrivilegesGranted = $dbTool->checkPrivileges($privileges);
+        $tablePrivilegesGranted = $dbTool->checkPrivileges($privileges, $tables);
+        if (! $dbPrivilegesGranted && ($isPgsql || ! $tablePrivilegesGranted)) {
+            return false;
+        }
+
+        if ($isPgsql && ! $tablePrivilegesGranted) {
```
[the remark this answers is not in the document]
Comment (yhabteab) [reply library/Icinga/Application/MigrationManager.php:348]: Yes! The user must have `CREATE,CONNECT,TEMPORARY` on a database level and `CREATE,USAGE` on a schema level and must be the owner of all tables.

File: modules/setup/application/forms/DbResourcePage.php [hunk 2]
```diff
@@ -141,14 +141,14 @@ protected function validateConfiguration()
...
         if ($this->getValue('db') === 'pgsql') {
             if ($connectionError !== null) {
-                $this->warning(sprintf(
-                    $this->translate('Unable to check the server\'s version. This is usually not a critical error'
-                    . ' as there is probably only access to the database permitted which does not exist yet. If you are'
-                    . ' absolutely sure you are running PostgreSQL in a version equal to or newer than 9.1,'
-                    . ' you can skip the validation and safely proceed to the next step. The error was: %s'),
-                    $connectionError->getMessage()
-                ));
-                $state = false;
```
[the remark this answers is not in the document]
Comment (yhabteab) [reply modules/setup/application/forms/DbResourcePage.php:151]: This causes the wizard not to elevate the database privileges and fail to create the tables automatically.

File: library/Icinga/Application/MigrationManager.php [hunk 3]
```diff
@@ -258,7 +258,9 @@ public function fixIcingaWebMysqlGrants(Sql\Connection $db, array $elevateConfig
         $tool = $this->createDbTool($db);
         $tool->connectToDb();
 
-        if ($tool->checkPrivileges(['SELECT'], [], $actualUsername)) {
+        $isPgsql = $db->getAdapter() instanceof Sql\Adapter\Pgsql;
+        // PgSQL CREATE,CONNECT, and TEMPORARY privileges on a database doesn't permit a user to read data from a table
```
[the remark this answers is not in the document]
Comment (yhabteab) [reply library/Icinga/Application/MigrationManager.php]: It's being checked only for `MySQL`.

**Pull request merged**